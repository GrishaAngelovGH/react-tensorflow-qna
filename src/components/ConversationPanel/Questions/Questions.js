import ListGroup from "react-bootstrap/ListGroup"

const questions = [
  "Which is the capital of the UK?",
  "What is the population of London?",
  "What is the area covered by London?",
  "How many boroughs does London have?",
  "What landmarks does London have?",
  "What climate does London have?",
  "How many languages are spoken in London?",
  "How about the infrastructure?",
  "What is the rail transport also known as?",
  "As a global financial center, what is it home to?"
]

const Questions = ({ onClick }) => (
  <ListGroup>
    {
      questions.map((v, i) => (
        <ListGroup.Item key={i} action onClick={() => { onClick(v) }}>
          {v}
        </ListGroup.Item>
      ))
    }
  </ListGroup>
)

export default Questions