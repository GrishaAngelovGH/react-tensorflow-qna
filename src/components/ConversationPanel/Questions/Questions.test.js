import { render } from '@testing-library/react'

import Questions from './Questions'

test('should render Questions component', () => {
  const view = render(<Questions />)

  expect(view).toMatchSnapshot()
})