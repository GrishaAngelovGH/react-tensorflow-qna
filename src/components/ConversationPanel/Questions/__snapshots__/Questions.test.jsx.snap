// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render Questions component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="list-group"
      >
        <button
          class="list-group-item list-group-item-action"
        >
          Which is the capital of the UK?
        </button>
        <button
          class="list-group-item list-group-item-action"
        >
          What is the population of London?
        </button>
        <button
          class="list-group-item list-group-item-action"
        >
          What is the area covered by London?
        </button>
        <button
          class="list-group-item list-group-item-action"
        >
          How many boroughs does London have?
        </button>
        <button
          class="list-group-item list-group-item-action"
        >
          What landmarks does London have?
        </button>
        <button
          class="list-group-item list-group-item-action"
        >
          What climate does London have?
        </button>
        <button
          class="list-group-item list-group-item-action"
        >
          How many languages are spoken in London?
        </button>
        <button
          class="list-group-item list-group-item-action"
        >
          How about the infrastructure?
        </button>
        <button
          class="list-group-item list-group-item-action"
        >
          What is the rail transport also known as?
        </button>
        <button
          class="list-group-item list-group-item-action"
        >
          As a global financial center, what is it home to?
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="list-group"
    >
      <button
        class="list-group-item list-group-item-action"
      >
        Which is the capital of the UK?
      </button>
      <button
        class="list-group-item list-group-item-action"
      >
        What is the population of London?
      </button>
      <button
        class="list-group-item list-group-item-action"
      >
        What is the area covered by London?
      </button>
      <button
        class="list-group-item list-group-item-action"
      >
        How many boroughs does London have?
      </button>
      <button
        class="list-group-item list-group-item-action"
      >
        What landmarks does London have?
      </button>
      <button
        class="list-group-item list-group-item-action"
      >
        What climate does London have?
      </button>
      <button
        class="list-group-item list-group-item-action"
      >
        How many languages are spoken in London?
      </button>
      <button
        class="list-group-item list-group-item-action"
      >
        How about the infrastructure?
      </button>
      <button
        class="list-group-item list-group-item-action"
      >
        What is the rail transport also known as?
      </button>
      <button
        class="list-group-item list-group-item-action"
      >
        As a global financial center, what is it home to?
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
